### DISCUSSION TITLE
[Project settings] [Environments] Edit environment entropy

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#65] [Ideas]
Hi guys,
I tried to edit existing environment from `stage` to `integration`.
I think that, at this moment, this feature is not useful, misleading, and brings a lot of entropy inside a project because:
- the k8s `namespace` label is not changed
- the environment variables PREFIX is not changed
- the environment id is not changed

In a real situation, after rename the `stage` environment to the `integration` environment of the project `my-tools`, the situation is the following:
- k8s namespace &rarr; `my-tools-stage`
- environment id &rarr; `stage`
- environment variables prefix &rarr; `STAGE_`
- environment shown by the Mia-Platform console &rarr; `integration`

In my opinion this feature should be completed or disabled.

**Comment by davidebianchi** [comment]
Hi @silversoul93, the rename of a environment is only by console side, so only for the name. All other information are more sensible to change without risks. And maybe there is not strong correlation between environment name and namespace (for example name of namespace of stage environment can be `my-project-preview`).

For example, rename a namespace in kubernetes bring you to delete and create a namespace. And if this namespace is production, and you don't know this, this bring a unexpected downtime.

If you want to rename all the environment info, also the runtime one, the correct flow is to delete and recreate the environment. This should be in the future easier, for example adding features already suggested like clone of public variables.

**Reply by frengooo** [reply]
Ok, it's more clear to me now and I agree that is dangerous to edit all during the rename.

Can we add an alert with this information in the `Edit Environment` page?

**Reply by davidebianchi** [reply]
In the edit you can see what you change, so the namespace does not change. It is not clear? I though that we should add in the docs how to migrate an environment (which has some manual steps). What do you think? Maybe with this documentation it would be clearer?

**Reply by frengooo** [reply]
I didn't notice that the namespace remained the same. Probably a couple of examples on real use cases will help to explain:
- when is appropriate to use the `Edit` feature
- when is appropriate to use the `Delete & Create` approach
